continuous_integration/scripts/install.sh

```diff
@@ -27,7 +27,7 @@ if [[ ${UPSTREAM_DEV} ]]; then
         git+https://github.com/dask/distributed \
         git+https://github.com/dask/dask-expr \
         git+https://github.com/dask/fastparquet \
-        git+https://github.com/zarr-developers/zarr-python.git@main
+        git+https://github.com/zarr-developers/zarr-python.git@fix/dask-compat
```

Comment:
I've added the `upstream` label to this PR which will now run a CI job with this branch from `zarr` when you push future commits here

Reply:
You're probably already aware, but just posing the current crop of CI failures here for convenience 

```
FAILED dask/array/tests/test_array_core.py::test_zarr_existing_array - TypeError: object of type 'generator' has no len()
FAILED dask/array/tests/test_array_core.py::test_zarr_pass_store - ValueError: mode mismatch (mode=a != store.mode=AccessMode(str='w', readonly=False, overwrite=True, create=True, update=False))
FAILED dask/array/tests/test_array_core.py::test_zarr_group - ValueError: store mode does not support writing
FAILED dask/array/tests/test_array_core.py::test_zarr_regions - TypeError: object of type 'generator' has no len()
FAILED dask/tests/test_order.py::test_array_store_final_order - AttributeError: module 'zarr' has no attribute 'DirectoryStore'
FAILED dask/tests/test_sizeof.py::test_xarray_not_in_memory - TypeError: open_group() takes 0 positional arguments but 1 positional argument (and 4 keyword-only arguments) were given
FAILED dask/tests/test_distributed.py::test_zarr_distributed_with_explicit_memory_store - FileNotFoundError: memory://140106500045760
FAILED dask/tests/test_distributed.py::test_zarr_in_memory_distributed_err - TypeError: object of type 'generator' has no len()
ERROR dask/tests/test_distributed.py::test_zarr_distributed_roundtrip - Failed: 3 thread(s) were leaked from test

------ Call stack of leaked thread 1/3: <Thread(zarrIO, started daemon 140107162052160)> ------
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1032, in _bootstrap
	self._bootstrap_inner()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
	self.run()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1012, in run
	self._target(*self._args, **self._kwargs)
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/asyncio/base_events.py", line 641, in run_forever
	self._run_once()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/asyncio/base_events.py", line 1948, in _run_once
	event_list = self._selector.select(timeout)
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/selectors.py", line 468, in select
	fd_event_list = self._selector.poll(timeout, max_ev)

------ Call stack of leaked thread 2/3: <Thread(asyncio_0, started daemon 140107587769920)> ------
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1032, in _bootstrap
	self._bootstrap_inner()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
	self.run()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1012, in run
	self._target(*self._args, **self._kwargs)
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/concurrent/futures/thread.py", line 89, in _worker
	work_item = work_queue.get(block=True)

------ Call stack of leaked thread 3/3: <Thread(asyncio_1, started daemon 140107178833472)> ------
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1032, in _bootstrap
	self._bootstrap_inner()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
	self.run()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1012, in run
	self._target(*self._args, **self._kwargs)
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/concurrent/futures/thread.py", line 89, in _worker
	work_item = work_queue.get(block=True)
ERROR dask/tests/test_distributed.py::test_zarr_distributed_with_explicit_directory_store - Failed: 2 thread(s) were leaked from test

------ Call stack of leaked thread 1/2: <Thread(asyncio_2, started daemon 140107145270848)> ------
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1032, in _bootstrap
	self._bootstrap_inner()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
	self.run()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1012, in run
	self._target(*self._args, **self._kwargs)
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/concurrent/futures/thread.py", line 89, in _worker
	work_item = work_queue.get(block=True)

------ Call stack of leaked thread 2/2: <Thread(asyncio_3, started daemon 140106375620160)> ------
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1032, in _bootstrap
	self._bootstrap_inner()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1075, in _bootstrap_inner
	self.run()
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/threading.py", line 1012, in run
	self._target(*self._args, **self._kwargs)
  File "/home/runner/miniconda3/envs/test-environment/lib/python3.12/concurrent/futures/thread.py", line 89, in _worker
	work_item = work_queue.get(block=True)
```